### PR TITLE
factor-lang: 0.98 -> 0.99

### PR DESCRIPTION
**Pull request by olus2000**
changelog: https://re.factorcode.org/2023/08/factor-0-99-now-available.html

## Description of changes

Bumped version of Factor language package to its newest release, 0.99. Modified patches to fit 0.99 file structure and changed source link.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by natsukium** [comment]
@ofborg eval

File: pkgs/development/compilers/factor-lang/factor99.nix [hunk 1]
```diff
@@ -219,6 +214,5 @@ stdenv.mkDerivation {
     license = licenses.bsd2;
     maintainers = with maintainers; [ vrthra spacefrogg ];
     platforms = lib.intersectLists platforms.x86_64 platforms.linux;
-    mainProgram = "factor";
```
Comment (spacefrogg) [inline pkgs/development/compilers/factor-lang/factor99.nix:222]: "factor" is still the main program.

**Review by spacefrogg — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/factor-lang/factor99.nix [hunk 2]
```diff
@@ -150,11 +150,6 @@ stdenv.mkDerivation {
       done)> $out/lib/factor/ld.so.cache
 
     make -j$NIX_BUILD_CORES linux-x86-64
-    printf "First build from upstream boot image\n" >&2
-    ./build.sh bootstrap
-    printf "Rebuild boot image\n" >&2
-    ./factor -script -e='"unix-x86.64" USING: system bootstrap.image memory ; make-image save 0 exit'
-    printf "Second build from local boot image\n" >&2
```
[the remark this answers is not in the document]
Comment (olus2000) [reply pkgs/development/compilers/factor-lang/factor99.nix]: I must have somehow cloned a nixpkgs state that didn't include your commit. Of course these snippets should not be removed.

File: pkgs/development/compilers/factor-lang/factor99.nix (hunk 1 above)
Comment (olus2000) [reply pkgs/development/compilers/factor-lang/factor99.nix]: I must have somehow cloned a nixpkgs state that didn't include your commit. Of course these snippets should not be removed.

**Review by spacefrogg — APPROVED** [review]
This looks good to me. Thank you!

**Pull request merged**